src/app/static/preview-question/preview-question.component.ts

```diff
@@ -24,30 +24,32 @@ export class PreviewQuestionComponent implements OnInit, AfterViewChecked {
   @Input() question: Question;
   @Input() tag: Tag;
 
+  @Output() change: EventEmitter<number> = new EventEmitter<number>();
+
   // stores associated tags that don't fit on the preview-question element to be shown in popover
   hiddenTags: Tag[];
 
   // represents how many tags from the tags can fit across the preview-question
   limit: number;
-
+  
   constructor(
     public router: Router,
     private questionIdService: QuestionService,
     private cdRef: ChangeDetectorRef,
   ) {}
 
   // On click of the 'View' button, it sets the question ID of the selected question
-  viewQuestion = selectQuestionId => {
-    // Stores selected question ID in the QuestionService for use in the ViewQuestionComponent
-    this.questionIdService.setQuestionId(selectQuestionId);
-    this.router.navigate([`/view-question/`]);
+  viewQuestion(selectedQuestionId) {
+    //Outputs id of selected question to parent.
+    this.change.emit(selectedQuestionId);
   }
 
   ngOnInit() {
     if (this.question.associatedTags) {
       this.limit = this.question.associatedTags.length;
     }
     this.hiddenTags = [];
+    this.questionIdService.setQuestionId(undefined);
```

Comment:
It's possible this is the source of all those 400s from the backend.  Why set questionId in the service to undefined with every preview-question component init?

Reply:
I don't even know what questionIdService is used for, so I just removed it completely from this component.